lib/lists.nix

```diff
@@ -312,11 +317,12 @@ rec {
        => { right = [ 5 3 4 ]; wrong = [ 1 2 ]; }
   */
   partition = builtins.partition or (pred:
-    foldr (h: t:
-      if pred h
-      then { right = [h] ++ t.right; wrong = t.wrong; }
-      else { right = t.right; wrong = [h] ++ t.wrong; }
-    ) { right = []; wrong = []; });
+    foldr
+      (h: t:
+        if pred h
+        then { right = [ h ] ++ t.right; wrong = t.wrong; }
+        else { right = t.right; wrong = [ h ] ++ t.wrong; }
+      ) { right = [ ]; wrong = [ ]; });
```

Comment:
I think because this is a partial function, the result should be:
```
partition = builtins.partition or (pred:
  foldr 
    (h: t:
      if pred h
      then { right = [ h ] ++ t.right; wrong = t.wrong; }
      else { right = t.right; wrong = [h] ++ t.wrong; }
    ) 
    { right = []; wrong = []; }
);
```